### PR TITLE
chore(db): modify state transition docs

### PR DESCRIPTION
**Pull request by rkrasiuk**
Modify the description and semantics of state transition table docs introduced in #430 and implemented in #449. Provide more examples for transition lookups.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm

File: docs/design/database.md [hunk 1]
```diff
@@ -45,24 +44,38 @@ TransactionHash {
...
-ChangeSet {
-    u64 ChangeId "PK"
-    ChangeSet PreviousValues "[Acc1[Balance,Nonce),Acc2(Balance,Nonce)] Previous values"
+    u64 TransitionId
 }
 History {
     H256 Account "PK"
-    u64 ChangeIdList "[ChangeId,ChangeId,...] Points where account changed"
+    u64 TransitionIdList "TransitionId[] (transitions where account changed)"
+}
```
Comment (rakita) [inline docs/design/database.md:58]: Same as we have two `ChangeSet` tables we have two `History` tables `AccountHistory` and `StorageHistory`

```
AccountHistory {
    H256 Account "PK"
    u64 TransitionIdList "TransitionId[] (transitions where account changed)"
}
StorageHistory {
    H256 Account "PK"
    H256 StorageKey "PK"
    u64 TransitionIdList "TransitionId[] (transitions where account changed)"
}
```

Comment (rkrasiuk) [reply docs/design/database.md:58]: good catch, fixing

**Comment by rkrasiuk** [comment]
@rakita i've modified the relations, tell me if that makes sense to you

**Comment by rakita** [comment]
nit: https://github.com/paradigmxyz/reth/blob/rkrasiuk/chore-transition-table-design/docs/design/database.md
DatabaseErr

**Comment by rkrasiuk** [comment]
interesting, mermaid.live was parsing the source just fine. anyway, fixed now

**Review by gakonst — APPROVED** [review]

**Comment by gakonst** [comment]
CI unrelated

**Pull request merged**